lib/request-base.js

```diff
@@ -155,17 +155,16 @@ RequestBase.prototype._retry = function() {
 
   // node
   if (this.req) {
-    delete this.req;
+    this.req = null;
+    this.req = this.request();
   }
 
   this._aborted = false;
   this.timedout = false;
   delete this._timer;
```

Comment:
In `_setTimeouts()` there is the following check, so if they are not removed, the new timers for retry can't be set:
```js
if (this._timeout && !this._timer) {
```
Probably better to delete or zero out in `clearTimeout()`?